docs/quickstart.fsx

```diff
@@ -0,0 +1,229 @@
+
+DiffSharp provides the most commonly used model building blocks including convolutions, transposed convolutions, batch normalization, dropout, recurrent and other architectures.
+
+See the [DiffSharp.Model](/reference/diffsharp-model.html) namespace for the full API reference.
+
+### Constructing models, PyTorch style
+
+If you have experience with [PyTorch](https://pytorch.org/), you would find the following way of model definition familiar. Let's look at an example of a [generative adversarial network (GAN)](https://arxiv.org/abs/1406.2661) architecture.
+*)
+open DiffSharp.Model
+open DiffSharp.Compose
+
+// PyTorch style
+
+// Define a model class inheriting the base
+type Generator(nz:int) =
+    inherit Model()
+    let fc1 = Linear(nz, 256)
+    let fc2 = Linear(256, 512)
+    let fc3 = Linear(512, 1024)
+    let fc4 = Linear(1024, 28*28)
+    do base.add([fc1; fc2; fc3; fc4])
+    override self.forward(x) =
+        x
+        |> dsharp.view([-1;nz])
+        |> fc1.forward
+        |> dsharp.leakyRelu(0.2)
+        |> fc2.forward
+        |> dsharp.leakyRelu(0.2)
+        |> fc3.forward
+        |> dsharp.leakyRelu(0.2)
+        |> fc4.forward
+        |> dsharp.tanh
+
+// Define a model class inheriting the base
+type Discriminator(nz:int) =
+    inherit Model()
+    let fc1 = Linear(28*28, 1024)
+    let fc2 = Linear(1024, 512)
+    let fc3 = Linear(512, 256)
+    let fc4 = Linear(256, 1)
+    do base.add([fc1; fc2; fc3; fc4])
+    override self.forward(x) =
+        x
+        |> dsharp.view([-1;28*28])
+        |> fc1.forward
+        |> dsharp.leakyRelu(0.2)
+        |> dsharp.dropout(0.3)
+        |> fc2.forward
+        |> dsharp.leakyRelu(0.2)
+        |> dsharp.dropout(0.3)
+        |> fc3.forward
+        |> dsharp.leakyRelu(0.2)
+        |> dsharp.dropout(0.3)
+        |> fc4.forward
+        |> dsharp.sigmoid
+
+// Instantiate the defined classes
+let nz = 128
+let gen = Generator(nz)
+let dis = Discriminator(nz)
+
+print gen
+print dis
+(*** include-output ***)
+
+(**
+### Constructing models, DiffSharp style
+
+DiffSharp's real strength lies in the [functional programming](https://en.wikipedia.org/wiki/Functional_programming) paradigm enabled by the F# language, where functions are first-class citizens, many algorithms can be constructed by applying and composing functions, and differentiation operations can be expressed as composable [higher-order functions](https://en.wikipedia.org/wiki/Higher-order_function). This allows very succinct (and beautiful) machine learning code to be expressed as a powerful combination of [lambda calculus](https://en.wikipedia.org/wiki/Lambda_calculus) and [differential calculus](https://en.wikipedia.org/wiki/Differential_calculus).
```

Comment:
"real" seems a little odd here.  It's an important strength, but I'd rate the nested differentiation capablities as more important